### ISSUE TITLE
New installation redirects to Laravel page? (self-hosted, Nginx)

### ISSUE THREAD

**Issue by srdco** [firefly-iii/firefly-iii#4076]
I am running Firefly III version 5.4.6.

**Description**
New installation per instructions on docs using composer on Ubuntu 18 LTS server running NGINX in a 'production' environment (personal/business webpages & apps)
Multiple other websites working just fine --- I'm guessing it's a configuration issue?

When access firefly-iii webpage (budget.example.com), get this output:
```php
. */ /** * Laravel - A PHP Framework For Web Artisans * * @package Laravel * @author Taylor Otwell */ define('LARAVEL_START', microtime(true)); /* |-------------------------------------------------------------------------- | Register The Auto Loader |-------------------------------------------------------------------------- | | Composer provides a convenient, automatically generated class loader for | our application. We just need to utilize it! We'll simply require it | into the script here so that we don't have to worry about manual | loading any of our classes later on. It feels great to relax. | */ require __DIR__ . '/../vendor/autoload.php'; /* |-------------------------------------------------------------------------- | Turn On The Lights |-------------------------------------------------------------------------- | | We need to illuminate PHP development, so let us turn on the lights. | This bootstraps the framework and gets it ready for use, then it | will load up this application so that we can run it and send | the responses back to the browser and delight our users. | */ /** @noinspection UsingInclusionOnceReturnValueInspection */ $app = require_once __DIR__ . '/../bootstrap/app.php'; /* |-------------------------------------------------------------------------- | Run The Application |-------------------------------------------------------------------------- | | Once we have the application, we can handle the incoming request | through the kernel, and send the associated response back to | the client's browser allowing them to enjoy the creative | and wonderful application we have prepared for them. | */ $kernel = $app->make(Illuminate\Contracts\Http\Kernel::class); $response = $kernel->handle( $request = Illuminate\Http\Request::capture() ); $response->send(); $kernel->terminate($request, $response);
```

**Extra info**
Accessed through Microsoft Windows 10, Edge (new = Chrome engine). No debug page available.

**Bonus points**
![Screenshot 2020-11-20 214138](https://user-images.githubusercontent.com/4694306/99866740-45fa4980-2b79-11eb-9b5d-fed1f6e5c4d8.png)
- I have read the FAQ at https://bit.ly/FF3-FAQ  -- YES
- I added a screenshot -- See attached

**Comment by srdco** [comment]
Here is the Nginx configuration:

`server {
    listen 80;
    server_name  budget.example.com;
    return 301 https://budget.example.com$request_uri;
}

server {
    include global/listen.conf;

    server_name budget.example.com;

    # Path to the root of your installation
    root /var/www/html/firefly-iii/public;
    index index.html index.htm index.php;

    #prevent HTTPoxy vulnerability
#    fastcgi_param HTTP_PROXY "";
 
    location / {
        try_files $uri /index.php$is_args$args;
        autoindex on;
        sendfile off;
    }

    location ~ \.php$ {
#        fastcgi_index index.php;
#        fastcgi_read_timeout 240;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
#        include fastcgi_params;
#        fastcgi_split_path_info ^(.+.php)(/.+)$;
#        proxy_set_header Host $http_host;
#        proxy_set_header X-Real_IP $remote_addr;
     }

#    location ~ /\.ht {
#        deny all;
#    }

    ssl_certificate /etc/letsencrypt/live/example.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/example.com/privkey.pem; # managed by Certbot

}`

**Comment by srdco** [comment]
Let's try that again:
```
server {
    listen 80;
    server_name  budget.fairviewhill.com;
    return 301 https://budget.fairviewhill.com$request_uri;
}

server {
    include global/listen.conf;

    server_name budget.fairviewhill.com;

    # Path to the root of your installation
    root /var/www/html/firefly-iii/public;
    index index.html index.htm index.php;

    #prevent HTTPoxy vulnerability
#    fastcgi_param HTTP_PROXY "";
 
    location / {
        try_files $uri /index.php$is_args$args;
        autoindex on;
        sendfile off;
    }

    location ~ \.php$ {
#        fastcgi_index index.php;
#        fastcgi_read_timeout 240;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
#        include fastcgi_params;
#        fastcgi_split_path_info ^(.+.php)(/.+)$;
#        proxy_set_header Host $http_host;
#        proxy_set_header X-Real_IP $remote_addr;
     }

#    location ~ /\.ht {
#        deny all;
#    }

    ssl_certificate /etc/letsencrypt/live/solidrock.enterprises/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/solidrock.enterprises/privkey.pem; # managed by Certbot

}
```

**Comment by srdco** [comment]
Wow ... this isn't my day. If there's a way to delete that last post so it doesn't have my server name, I'd appreciate it. That's what I get for cut and pasting without re-reading.

**Comment by death2cupbots** [comment]
Is there a reason most of the stuff under your php section are commented out?
Right now my php section looks like:
```
        location ~ \.php$ {
              try_files $uri =404;
              fastcgi_pass unix:/var/run/php/php7.4-fpm.sock;
              fastcgi_index index.php;
              fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
              include fastcgi_params;

        }
```
From what I can tell, it doesn't actually look like you're telling firefly iii where your php is so it can use it.

**Comment by JC5** [comment]
There's an edit button next to each comment, but I've taken the liberty and I've edited your posts.

**Comment by srdco** [comment]
JC5 -- thanks. I looked for it because I've had to use it before ... but for some reason it wasn't showing up last night.

**Comment by srdco** [comment]
> Is there a reason most of the stuff under your php section is commented out?

It was commented out because I was trying removing one line at a time to see if anything was conflicting. It did the same thing even with everything uncommented.

**Comment by srdco** [comment]
So based on your comment @death2cupbots, I scrutinized it again this morning (I should know better than to try and spin up a new web app AFTER work!)

It turned out, I was confusing NGINX by having the generic "index" set. Once I removed that line, it stopped trying to load the laravel page. Once I fixed the PHP section (I just took the easy shortcut and used yours since it was working!) everything works great. Thanks!
